### PR TITLE
Workaround for GHC 8.10.5 on macOS

### PR DESCRIPTION
**Pull request by konn**
As pointed out by @mouse07410 in #1913, we can build HLS with GHC 8.10.5 on macOS just by removing CommandLineTools from the environment.
This PR tries to adopt this workaround in a rather ad-hoc manner.

### CODE REVIEW

**Comment by jneira** [comment]
> 
> 
> I want to test build action for GHC 8.10.5 + macOS - is there any way to invoke the build workflow manually?

It is triggered on releases so i usually test them in my repo, creating a fake release there and delete it after the test.
(We could change the trigger in the script and restore it after the test but i would recommend test it in your repo, you could link to your succesful build for completeness)

**Comment by konn** [comment]
Here it goes! :rocket: https://github.com/konn/haskell-language-server/runs/2837481049?check_suite_focus=true

**Comment by konn** [comment]
Hurray! It at least builds the `network` with GHC 8.10.5 on macOS 🎉

スクリーンショット 2021-06-16 18 02 31

**Comment by Ailrun** [comment]
Awesome! Now we can have GHC 8.10.5 on macOS for future versions too :)

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -74,6 +72,12 @@ jobs:
       run: |
         echo "GHC_VERSION=$GHC_VER" >> $GITHUB_ENV
 
+    - name: Tentative Workaround for GHC 8.10.5 on macOS
+      if: matrix.os == 'macOS-latest' && matrix.ghc == '8.10.5'
+      run: |
+        echo "# uninstalling CommandLineTools (see https://github.com/haskell/haskell-language-server/issues/1913#issuecomment-861667786)"
+        sudo rm -rf /Library/Developer/CommandLineTools
+
```
Comment (Ailrun) [inline .github/workflows/build.yml:80]: Could you also apply the similar trick to test CI so that we can test 8.10.5 build on macOS for each PR?

Comment (konn) [reply .github/workflows/build.yml:80]: Good point! Added to test.yml.

**Comment by konn** [comment]
OK, it finally generated HLS binary for GHC 8.10.5 on macOS 🎉

**Comment by konn** [comment]
I will re-upload the HLS exes generated by GHA for macOS to haskell/haskell-language-server after the test CI successfully passes.

**Review by jneira — APPROVED** [review]
Amazing work, many thanks, macOS users will be grateful

**Comment by konn** [comment]
The hack incoporated in this pull request is essentially due to @mouse07410.
Credit goes to them, great discovery!

**Comment by konn** [comment]
Oops, tests got cancelled by ghc 9 failure -- is this intended behaviour?

**Comment by konn** [comment]
Ah, I used the wrong commit to build 1.2.0 on my fork repository. This Pull Request itself is correct, but I will re-run the dummy release on my fork to generate macOS binaries with the correct commit.

**Comment by konn** [comment]
I've just re-uploaded gzip and tarballs, as it seems all the tests pass now.

**Pull request merged**